schemas/2015-11-01/Microsoft.Network.json

```diff
@@ -0,0 +1,159 @@
+                        },
+                        "dnsConfig": {
+                            "type": "object",
+                            "properties": {
+                                "relativeName": {
+                                    "type": "string"
+                                },
+                                "ttl": {
+                                    "type": "integer",
+                                    "minimum": 30,
+                                    "maximum": 2147483647
+                                },
+                                "fqdn": {
+                                    "type": "string"
+                                },
+                                "required": [
+                                    "relativeName",
+                                    "ttl"
+                                ]
+                            }
+                        },
+                        "monitorConfig": {
+                            "type": "object",
+                            "properties": {
+                                "protocol": {
+                                    "enum": [
+                                        "HTTP",
+                                        "HTTPS"
+                                    ]
+                                },
+                                "port": {
+                                    "type": "integer",
+                                    "minimum": 1,
+                                    "maximum": 65535
+                                },
+                                "path": {
+                                     "type": "string"
+                                }
+                            },
+                            "required": [
+                                "protocol",
+                                "port",
+                                "path"
+                            ]
+                        },
+                        "endpoints": {
+                            "type": "array",
+                            "items": {
+                                "type": "object",
+                                "properties": {
+                                    "name": {
+                                        "type": "string"
+                                    },
+                                    "type": {
+                                         "enum": [
+                                              "Microsoft.Network/trafficManagerProfiles/azureEndpoints",
+                                              "Microsoft.Network/trafficManagerProfiles/externalEndpoints",
+                                              "Microsoft.Network/trafficManagerProfiles/nestedEndpoints"
+                                        ]
+                                    },
+                                    "properties": {
+                                        "type": "object",
+                                        "properties": {
```

Comment:
some properties are required but which of them are depends on "trafficRoutingMethod" and "endpoints/type"; none of these properties are required in any and all cases